From 793a74065926b0da87120d4b1e6330234475505c Mon Sep 17 00:00:00 2001
From: Godbach <nylzhaowei@gmail.com>
Date: Tue, 9 Jun 2015 19:41:52 +0800
Subject: [PATCH 17/18] CLEANUP: deinit: remove codes for cleaning
 p->block_rules

Since all rules listed in p->block_rules have been moved to the beginning of
the http-request rules in check_config_validity(), there is no need to clean
p->block_rules in deinit().

Signed-off-by: Godbach <nylzhaowei@gmail.com>
(cherry picked from commit 28b48ccbc879a552f988e6e1db22941e3362b4db)
---
 src/haproxy.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/haproxy.c b/src/haproxy.c
index 0dddd53..eac6f44 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -1020,12 +1020,6 @@ void deinit(void)
 			free(cwl);
 		}
 
-		list_for_each_entry_safe(cond, condb, &p->block_rules, list) {
-			LIST_DEL(&cond->list);
-			prune_acl_cond(cond);
-			free(cond);
-		}
-
 		list_for_each_entry_safe(cond, condb, &p->mon_fail_cond, list) {
 			LIST_DEL(&cond->list);
 			prune_acl_cond(cond);
-- 
2.3.6

